homeassistant/components/network/util.py

```diff
@@ -0,0 +1,179 @@
+"""Network helper class for the network integration."""
+from __future__ import annotations
+
+from collections.abc import Iterable
+from ipaddress import IPv4Address, IPv6Address, ip_address
+import logging
+from typing import cast
+
+import ifaddr
+from pyroute2 import IPRoute
+
+from homeassistant.core import HomeAssistant
+
+from .const import MDNS_TARGET_IP
+from .models import Adapter, IPv4ConfiguredAddress, IPv6ConfiguredAddress
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_load_adapters(hass: HomeAssistant) -> list[Adapter]:
+    """Load adapters."""
+    next_hop = await _async_default_next_broadcast_hop(hass)
```

Comment:
Maybe we can avoid pyroute2 usage here by opening a socket to the next hop and checking the local address ?  Need to test that

Reply:
If that works it would be more portable

Reply:
We could also provide a function for integrations like dlna to find the right interface to bind so it's reachable as listen_ip isn't being detected per the url

Reply:
The functional test (non-blocking) worked as expected and should be more portable. The `pyroute2` requirement has been removed.